packages/workflow/src/records/fhir.ts

```diff
@@ -1357,46 +1483,235 @@ function toSavedComposition(
   }
 }
 
+function toSavedTask(
+  task: Task & { focus: { reference: URNReference } },
+  id: UUID,
+  resourceBundle: Bundle,
+  responseBundle: TransactionResponse
+): SavedTask {
+  const savedReference = findSavedReference(
+    task.focus.reference,
+    resourceBundle,
+    responseBundle
+  )
+  if (!savedReference) {
+    throw internal(
+      `No response found for "task.focus.reference->${
+        task.focus.reference
+      }" in the following transaction: ${JSON.stringify(responseBundle)}`
+    )
+  }
+  return {
+    ...task,
+    id,
+    focus: {
+      reference: urlReferenceToResourceIdentifier(savedReference)
+    }
+  }
+}
+
+function toSavedRelatedPerson(
+  relatedPersion: RelatedPerson & {
+    patient: { reference: `urn:uuid:${string}` }
+  },
+  id: UUID,
+  resourceBundle: Bundle,
+  responseBundle: TransactionResponse
+): SavedRelatedPerson {
+  const savedReference = findSavedReference(
+    relatedPersion.patient.reference,
+    resourceBundle,
+    responseBundle
+  )
+  if (!savedReference) {
+    throw internal(
+      `No response found for "relatedPersion.patient.reference->${
+        relatedPersion.patient.reference
+      }" in the following transaction: ${JSON.stringify(responseBundle)}`
```

Comment:
Could this error leak PII, if you refer to this comment: https://github.com/opencrvs/opencrvs-core/issues/6419#issuecomment-1909964775

Reply:
It shouldn't leak any PII as the data I'm logging only contains references i.e. UUIDs